homework-submission/todoapp/src/Components/TodoElement.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { observer, inject } from 'mobx-react';
+
+@inject('todos')
+@observer
+class TodoItem extends React.Component {
+  state = {
+    edit: false
+  }
+  toggleHandler = () => {
+    this.setState({
+      edit: !this.state.edit
+    })
+  }
+  handleUpdate = (e) => {
+    e.preventDefault()
+    const todo = this.props.todo
+    const { updateHandler } = this.props.todos
+    let newTask = this.editField.value
+    updateHandler(todo._id, newTask)
```

Comment:
After the update is happening the form is not switching to be the list item by it self I assume you missed a `setState` here? or a toggle?